examples/ongoing/src/main/java/me/digi/examples/ongoing/ui/home/BreakdownAdapter.kt

```diff
@@ -0,0 +1,69 @@
+package me.digi.examples.ongoing.ui.home
+
+import android.content.Context
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.core.content.contentValuesOf
+import androidx.recyclerview.widget.RecyclerView
+import io.objectbox.Box
+import kotlinx.android.synthetic.main.genre_breakdown_item.view.*
+import me.digi.examples.ongoing.model.Genre
+import me.digi.examples.ongoing.model.Song
+import me.digi.examples.ongoing.service.ObjectBox
+import me.digi.examples.ongoing.utils.GenreInsightGenerator
+import me.digi.ongoing.R
+
+class BreakdownAdapter(val context: Context) : RecyclerView.Adapter<BreakdownAdapter.ViewHolder>() {
+
+    private val genreBox: Box<Song> = ObjectBox.boxStore.boxFor(Song::class.java)
+    private val genreInsights = GenreInsightGenerator.generateGenrePair(genreBox.query().build().find())
+
+    inner class ViewHolder(item: View) : RecyclerView.ViewHolder(item) {
+        private val genreName = itemView.genreLabel
+        private val playCount = itemView.songCount
+        private val percentBar = itemView.percentBar
+
+        private fun setPercentBarColor() {
+            val randomColourIndex = (0 until 12).random()
+            val colours = listOf(
```

Comment:
If you move this to be a property on a companion object of `ViewHolder` it will only be computed once, as opposed to for each item.